src/ui/map.js

```diff
@@ -2522,13 +2522,14 @@ class Map extends Camera {
         if (somethingDirty || this._repaint) {
             this.triggerRepaint();
         } else if (!this.isMoving() && this.loaded()) {
-            if (!this._fullyLoaded) {
-                this._fullyLoaded = true;
-                PerformanceUtils.mark(PerformanceMarkers.fullLoad);
-            }
             this.fire(new Event('idle'));
         }
 
+        if (!this._fullyLoaded) {
```

Comment:
I think this check needs to be `this._loaded && !this._fullyLoaded && !somethingDirty`

Reply:
fixed